include/fastrtps/types/DynamicPubSubType.h

```diff
@@ -45,6 +45,8 @@ class DynamicPubSubType : public eprosima::fastdds::dds::TopicDataType
 
 public:
 
+    using type = DynamicData_ptr;
```

Comment:
```suggestion
    typedef DynamicData type;
```

Reply:
I think doing this will allow using the methods in the PubSubWriter, instead of using the `native_writer()`

Reply:
Thanks @MiguelCompany for helping me with 94adf29! In the end this suggestions makes the test itself much cleaner and brings an even cooler infrastructure to the PubSubReader and Writer but was not so straight forward 🤣